controller/main.go

```diff
@@ -136,6 +138,14 @@ func (c *controller) SetPools(l log.Logger, pools *config.Pools) controllers.Syn
 	return controllers.SyncStateReprocessAll
 }
 
+func (c *controller) SetNode(l log.Logger, node *v1.Node) controllers.SyncState {
+	if k8snodes.IsNodeExcludedFromBalancers(node) {
+		return controllers.SyncStateSuccess
+	}
+	c.nodes = append(c.nodes, *node)
```

Comment:
Pool reject at webhook validation is more straight forward and earlier than going into pool controller. Since it will reject right away with error message:
kubectl apply -f ipAddressPool_overlap.yaml
Error from server (Forbidden): error when creating "ipAddressPool_overlap.yaml": admission webhook "ipaddresspoolvalidationwebhook.metallb.io" denied the request: pool cidr "172.18.6.0/24" contains nodeIp "172.18.6.2"

Reply:
Ok, I can check and prevent duplicates. But dangling deleted node should not be an issue since the node might be added back later and we should still prevent service to use an IP that was used by a node before.